langchain4j-azure-open-ai/pom.xml

```diff
@@ -21,11 +21,128 @@
             <artifactId>langchain4j-core</artifactId>
         </dependency>
 
+        <!-- DEPENDENCY CONFLICT RESOLUTION FOR AZURE-AI-OPENAI (START) -->
+
+        <!-- To resolve version conflicts inside the 'azure-ai-openai' library, we are excluding the transitive -->
```

Comment:
Is this causing a dependency conflict with other Azure libraries? Could you please share the `mvn dependency:tree`?

Reply:
There is the result of `mvn dependency:tree` before above changes were applied:
```
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] -------------< dev.langchain4j:langchain4j-azure-open-ai >--------------
[INFO] Building LangChain4j :: Integration :: Azure OpenAI 0.33.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- dependency:3.6.1:tree (default-cli) @ langchain4j-azure-open-ai ---
[INFO] dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[INFO] +- dev.langchain4j:langchain4j-core:jar:0.33.0-SNAPSHOT:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.10.1:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:2.0.7:compile
[INFO] +- com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[INFO] |  +- com.azure:azure-core:jar:1.49.1:compile
[INFO] |  |  +- com.azure:azure-xml:jar:1.0.0:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.16.1:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.16.1:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.16.1:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.16.1:compile
[INFO] |  |  \- io.projectreactor:reactor-core:jar:3.4.38:compile
[INFO] |  |     \- org.reactivestreams:reactive-streams:jar:1.0.4:compile
[INFO] |  \- com.azure:azure-core-http-netty:jar:1.15.1:compile
[INFO] |     +- io.netty:netty-handler:jar:4.1.110.Final:compile
[INFO] |     |  +- io.netty:netty-resolver:jar:4.1.110.Final:compile
[INFO] |     |  \- io.netty:netty-transport:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[INFO] |     |  \- io.netty:netty-codec-socks:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-buffer:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-codec:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-codec-http:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[INFO] |     |  \- io.netty:netty-transport-classes-epoll:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[INFO] |     |  \- io.netty:netty-transport-classes-kqueue:jar:4.1.110.Final:compile
[INFO] |     +- io.netty:netty-tcnative-boringssl-static:jar:2.0.65.Final:compile
[INFO] |     |  +- io.netty:netty-tcnative-classes:jar:2.0.65.Final:compile
[INFO] |     |  +- io.netty:netty-tcnative-boringssl-static:jar:linux-x86_64:2.0.65.Final:compile
[INFO] |     |  +- io.netty:netty-tcnative-boringssl-static:jar:linux-aarch_64:2.0.65.Final:compile
[INFO] |     |  +- io.netty:netty-tcnative-boringssl-static:jar:osx-x86_64:2.0.65.Final:compile
[INFO] |     |  +- io.netty:netty-tcnative-boringssl-static:jar:osx-aarch_64:2.0.65.Final:compile
[INFO] |     |  \- io.netty:netty-tcnative-boringssl-static:jar:windows-x86_64:2.0.65.Final:compile
[INFO] |     +- io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[INFO] |     |  +- io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[INFO] |     |  |  \- io.netty:netty-codec-dns:jar:4.1.109.Final:compile
[INFO] |     |  +- io.netty:netty-resolver-dns-native-macos:jar:osx-x86_64:4.1.109.Final:compile
[INFO] |     |  |  \- io.netty:netty-resolver-dns-classes-macos:jar:4.1.109.Final:compile
[INFO] |     |  \- io.projectreactor.netty:reactor-netty-core:jar:1.0.45:compile
[INFO] |     \- io.netty:netty-common:jar:4.1.110.Final:compile
[INFO] +- com.azure:azure-identity:jar:1.13.1:compile
[INFO] |  +- com.azure:azure-json:jar:1.1.0:compile
[INFO] |  +- com.microsoft.azure:msal4j:jar:1.16.1:compile
[INFO] |  |  +- com.nimbusds:oauth2-oidc-sdk:jar:11.9.1:compile
[INFO] |  |  |  +- com.github.stephenc.jcip:jcip-annotations:jar:1.0-1:compile
[INFO] |  |  |  +- com.nimbusds:content-type:jar:2.3:compile
[INFO] |  |  |  +- com.nimbusds:lang-tag:jar:1.7:compile
[INFO] |  |  |  \- com.nimbusds:nimbus-jose-jwt:jar:9.37.3:compile
[INFO] |  |  \- net.minidev:json-smart:jar:2.5.0:compile
[INFO] |  |     \- net.minidev:accessors-smart:jar:2.5.0:compile
[INFO] |  |        \- org.ow2.asm:asm:jar:9.3:compile
[INFO] |  +- com.microsoft.azure:msal4j-persistence-extension:jar:1.3.0:compile
[INFO] |  |  \- net.java.dev.jna:jna:jar:5.13.0:compile
[INFO] |  \- net.java.dev.jna:jna-platform:jar:5.6.0:compile
[INFO] +- dev.langchain4j:langchain4j-core:test-jar:tests:0.33.0-SNAPSHOT:test
[INFO] +- com.knuddels:jtokkit:jar:1.1.0:compile
[INFO] +- org.apache.logging.log4j:log4j-api:jar:2.22.0:test
[INFO] +- org.apache.logging.log4j:log4j-core:jar:2.22.0:test
[INFO] +- org.apache.logging.log4j:log4j-slf4j2-impl:jar:2.20.0:test
[INFO] +- org.junit.jupiter:junit-jupiter-engine:jar:5.10.0:test
[INFO] |  +- org.junit.platform:junit-platform-engine:jar:1.10.0:test
[INFO] |  |  +- org.opentest4j:opentest4j:jar:1.3.0:test
[INFO] |  |  \- org.junit.platform:junit-platform-commons:jar:1.10.0:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-api:jar:5.10.0:test
[INFO] |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
[INFO] +- org.junit.jupiter:junit-jupiter-params:jar:5.10.0:test
[INFO] \- org.assertj:assertj-core:jar:3.24.2:test
[INFO]    \- net.bytebuddy:byte-buddy:jar:1.14.10:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.426 s
[INFO] Finished at: 2024-08-13T07:57:49+02:00
[INFO] ------------------------------------------------------------------------
```

But more valuable info might be the error report of maven enforcer plugin (when enabled):
```
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO]
[INFO] -------------< dev.langchain4j:langchain4j-azure-open-ai >--------------
[INFO] Building LangChain4j :: Integration :: Azure OpenAI 0.33.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- enforcer:3.5.0:enforce (enforce) @ langchain4j-azure-open-ai ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.705 s
[INFO] Finished at: 2024-08-13T08:00:07+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.5.0:enforce (enforce) on project langchain4j-azure-open-ai:
[ERROR] Rule 0: org.apache.maven.enforcer.rules.dependency.DependencyConvergence failed with message:
[ERROR] Failed while enforcing releasability.
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.projectreactor.netty:reactor-netty-core:jar:1.0.45:compile
[ERROR]           +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-transport-native-unix-common:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-epoll:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-kqueue:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns-native-macos:jar:osx-x86_64:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-resolver-dns-classes-macos:jar:4.1.109.Final:compile
[ERROR]             +-io.netty:netty-transport-native-unix-common:jar:4.1.109.Final:compile
[ERROR] 
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-codec-http2:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-codec-http2:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-transport:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec-socks:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-epoll:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-kqueue:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-transport:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-codec-dns:jar:4.1.109.Final:compile
[ERROR]             +-io.netty:netty-transport:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-common:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-resolver:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec-socks:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-epoll:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-kqueue:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-common:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-codec-dns:jar:4.1.109.Final:compile
[ERROR]             +-io.netty:netty-common:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns-native-macos:jar:osx-x86_64:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-resolver-dns-classes-macos:jar:4.1.109.Final:compile
[ERROR]             +-io.netty:netty-common:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-common:jar:4.1.110.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-resolver:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-resolver:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-resolver:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-resolver:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-handler-proxy:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.projectreactor.netty:reactor-netty-core:jar:1.0.45:compile
[ERROR]           +-io.netty:netty-handler-proxy:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-codec-http:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-codec-http:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-codec:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec-socks:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-codec:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-codec-dns:jar:4.1.109.Final:compile
[ERROR]             +-io.netty:netty-codec:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-handler:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-handler:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.projectreactor.netty:reactor-netty-core:jar:1.0.45:compile
[ERROR]           +-io.netty:netty-handler:jar:4.1.109.Final:compile
[ERROR]
[ERROR]
[ERROR] Dependency convergence error for io.netty:netty-buffer:jar:4.1.110.Final paths to dependency are:
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-handler-proxy:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-codec-socks:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-codec-http2:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-unix-common:jar:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-epoll:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.110.Final:compile
[ERROR]         +-io.netty:netty-transport-classes-kqueue:jar:4.1.110.Final:compile
[ERROR]           +-io.netty:netty-buffer:jar:4.1.110.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-buffer:jar:4.1.109.Final:compile
[ERROR] and
[ERROR] +-dev.langchain4j:langchain4j-azure-open-ai:jar:0.33.0-SNAPSHOT
[ERROR]   +-com.azure:azure-ai-openai:jar:1.0.0-beta.10:compile
[ERROR]     +-com.azure:azure-core-http-netty:jar:1.15.1:compile
[ERROR]       +-io.projectreactor.netty:reactor-netty-http:jar:1.0.45:compile
[ERROR]         +-io.netty:netty-resolver-dns:jar:4.1.109.Final:compile
[ERROR]           +-io.netty:netty-codec-dns:jar:4.1.109.Final:compile
[ERROR]             +-io.netty:netty-buffer:jar:4.1.109.Final:compile
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

So as you can see there are dependency conflict inside `com.azure:azure-ai-openai` (more precisely inside `com.azure:azure-core-http-netty`).

So in short - the code block inside `<!-- DEPENDENCY CONFLICT RESOLUTION FOR AZURE-AI-OPENAI (START) -->` of this pull request resolves these conflicts by using the highest specified verions of the conflicting libraries.